src/signal.h

```diff
@@ -26,6 +26,14 @@ void signal_set_handlers();
 /// default action (SIG_DFL) will be set
 void signal_handle(int sig, int do_handle);
 
+#define signal_block()
+#define signal_unblock()
+#define signal_is_blocked() false
+#if 0
+// TODO: Determine if the following three functions will need to be reinstated. For the time being
+// they are #define stubs since they appear to be unnecessary and noticably slow the shell down.
+// Be sure to reinstate the CHECK_BLOCK macro if these functions are reinstated.
+
```

Comment:
It makes me nervous as well. However, the existing implementation to block/unblock every signal is clearly akin to firing a shotgun to avoid being bitten by a mosquito. This is why I implemented this by implementing no-op stubs rather than removing the calls to these functions. We can reinstate the original version quickly without having to worry about resolving change conflicts.